evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/EvoForm.tsx

```diff
@@ -0,0 +1,54 @@
+import TextField from "@mui/material/TextField";
+import { useForm } from "../../../hooks/use-form";
+
+export type FormFieldAttr = {
+  label: string;
+  type: string;
+  value: string;
+};
+type FormProps = {
+  disable: boolean;
+  initialData: FormFieldAttr[];
+  callback: any;
+  buttonText: string;
+};
+
+export const EvoForm = ({
+  disable,
+  initialData,
+  callback,
+  buttonText,
+}: FormProps) => {
+  const data = initialData?.reduce(
+    (a, v) => ({ ...a, [v.label]: v.value }),
+    {}
+  );
+  const { onChange, onSubmit, values } = useForm(callback, data);
+
+  return (
+    <>
+      <form onSubmit={onSubmit}>
+        {initialData?.map((field) => (
+          <div>
+            <TextField
```

Comment:
All our fields are text? What about number/int etc.?

Reply:
it's just a wrapper to field , you can configure the type with "type" attribute , and i'm not sure we stay we this component maybe better to use <input> instead,,, we decide when we define the design